pulp_ansible/app/tasks/collections.py

```diff
@@ -343,15 +343,22 @@ def _build_url(path_or_url):
             downloader = remote.get_downloader(url=url)
             initial_data = parse_metadata(await downloader.run())
 
-            count = len(self.collection_info) or initial_data.get("count", 1)
+            api_version = None
+            if "results" in initial_data.keys():
```

Comment:
I think inspecting the remote's url would give us a more reliable case when future versions use one of these two response styles and yet we need to detect v4 for example.

Also can we make api_version and integer so simplicity.

That would make this block:

```
api_version = None
if "/api/v2/" in remote.url:
    api_version = 2
if "/api/v3/" in remote.url:
    api_version

Another option would be to regexparse the url itself and match on the integer.

```
import regex
galaxy_version_regex_compiled = re.compile('/api/v(\d)')
match_result = galaxy_version_regex_compiled.match(remote.url)
if match_result:
    api_version = int(match_result.groups()[0])
else:
    raise RuntimeError('Could not determine Galaxy API version')
```

Reply:
I thought the deduplication idea was to avoid  these string comparisons such as `"/api/v2/"`

Reply:
I believe the idea was to reuse the api_version after parsing it from the url. So this:

```python
# v2 uses 'results' as the key while v3 uses 'data'
results = data.get("results") or data.get("data") or [data]
```

Would become:

```python
if api_version == 2:
        results = data.get("results", [data])
else:
    results = data.get("data", [data])
```

Reply:
I think the current code is more pythonic

Reply:
My believe is that python prefers explicitness over implicitness. For example the need to add an additional comment suggests the code itself isn't very readable on its own basis.

Reply:
the URL approach fails for `PulpToPulpSyncCase`:
```
(pulp) (pulp) [vagrant@pulp3-source-fedora31 pulp_ansible]$ nc 127.0.0.1 4444
> /home/vagrant/devel/pulp_ansible/pulp_ansible/app/tasks/collections.py(345)_fetch_collections()
-> url = _get_url(page_count, api_version)
(Pdb) api_version
2
(Pdb) remote.url
'https://galaxy.ansible.com/api/v2/collections/testing/k8s_demo_collection'
(Pdb) c




^C
(pulp) (pulp) [vagrant@pulp3-source-fedora31 pulp_ansible]$ nc 127.0.0.1 4444
Ncat: Connection refused.
(pulp) (pulp) [vagrant@pulp3-source-fedora31 pulp_ansible]$ nc 127.0.0.1 4444
> /home/vagrant/devel/pulp_ansible/pulp_ansible/app/tasks/collections.py(345)_fetch_collections()
-> url = _get_url(page_count, api_version)
(Pdb) remote.url
'http://pulp3-source-fedora31.lan/pulp_ansible/galaxy/55ec1549-1a44-49ec-b10f-b96cd27c31da/api/v3/collections/testing/k8s_demo_collection'
(Pdb) api_version
3
```